pkg/internal/docker/start_jenkins.go

```diff
@@ -0,0 +1,159 @@
+func (o *RunOptions) CreateImageAndRunContainer(cmd *cobra.Command, args []string) (err error) {
+	ctx := context.Background()
+	cli, err := o.ConnectToDocker()
+	if err != nil {
+		cmd.Println(err)
+		return err
+	}
+	imageName := o.ImageName + ":" + o.Tag
+	err = o.BuildImage(cmd)
+	if err != nil {
+		cmd.Println(err)
+	}
+	jenkinsPort, err := nat.NewPort("tcp", "8080")
+	if err != nil {
+		cmd.Println(err)
+	}
+	hostConfig := &container.HostConfig{
+		PortBindings: nat.PortMap{
+			jenkinsPort: []nat.PortBinding{
+				{
+					HostIP:   "0.0.0.0",
+					HostPort: strconv.Itoa(o.JenkinsPort),
+				},
+			},
+		},
+	}
+	exposedPorts := map[nat.Port]struct{}{
+		jenkinsPort: struct{}{},
+	}
+	config := &container.Config{
+		Image:        imageName,
+		ExposedPorts: exposedPorts,
+	}
+
+	resp, err := cli.ContainerCreate(ctx, config, hostConfig, nil, nil, "jenkinstest2")
+	if err != nil {
+		fmt.Println(err)
+	}
+	cli.ContainerStart(ctx, resp.ID, types.ContainerStartOptions{})
+	fmt.Println(resp.ID)
+	return nil
+}
+
+//BuildImage is used to build an image
+func (o *RunOptions) BuildImage(cmd *cobra.Command) error {
+	ctx := context.Background()
+	cli, _ := o.ConnectToDocker()
+	dockerFileTarReader, _ := o.GetTarReader(cmd)
+	buildOptions := types.ImageBuildOptions{
+		Context:    dockerFileTarReader,
+		Dockerfile: o.DockerfilePath,
+		Remove:     true,
+		Tags:       []string{o.ImageName},
+	}
+	imageBuildResponse, err := cli.ImageBuild(
+		ctx,
+		dockerFileTarReader,
+		buildOptions,
+	)
+
+	if err != nil {
+		return err
+	}
+	defer imageBuildResponse.Body.Close()
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(imageBuildResponse.Body)
+	cmd.Println(buf.String())
+	return nil
+}
+
+//GetTarReader creates tarReader for args in function BuildImage
+func (o *RunOptions) GetTarReader(cmd *cobra.Command) (*bytes.Reader, error) {
+	src := []string{o.DockerfilePath, o.WarPath, "/tmp/jenkins-cli-docker.sh"}
+	buf := new(bytes.Buffer)
+	tw := tar.NewWriter(buf)
+	defer tw.Close()
+
+	for _, fileName := range src {
+		dockerFileReader, err := os.Open(fileName)
+		if err != nil {
+			return nil, err
+		}
+		readDockerFile, err := ioutil.ReadAll(dockerFileReader)
+		if err != nil {
+			return nil, err
+		}
+		tarHeader := &tar.Header{
+			Name: fileName,
+			Size: int64(len(readDockerFile)),
+		}
+		err = tw.WriteHeader(tarHeader)
+		if err != nil {
+			return nil, err
+		}
+		_, err = tw.Write(readDockerFile)
+		if err != nil {
+			return nil, err
+		}
+	}
+	dockerFileTarReader := bytes.NewReader(buf.Bytes())
+	return dockerFileTarReader, nil
+}
+
+//CreateDockerfile will create a docker file for running a jenkins war contains plugins hpi
+func (o *RunOptions) CreateDockerfile(cmd *cobra.Command, args []string) (err error) {
+	sh := "java -Djenkins.install.runSetupWizard=false -jar /usr/share/jenkins/jenkins.war\n" +
+		"tail -f /dev/null"
+	err = ioutil.WriteFile("/tmp/jenkins-cli-docker.sh", []byte(sh), 0)
```

Comment:
Please use `ioutil.TempFile` to create a temporary files, including temporary Dockerfile.